pgrx-macros/src/lib.rs

```diff
@@ -836,11 +836,12 @@ fn impl_postgres_type(ast: DeriveInput) -> syn::Result<proc_macro2::TokenStream>
             }
 
             #[doc(hidden)]
-            #[::pgrx::pgrx_macros::pg_extern(immutable,parallel_safe)]
-            pub fn #funcname_out #generics(input: #name #generics) -> &#lifetime ::core::ffi::CStr {
+            #[::pgrx::pgrx_macros::pg_extern (immutable,parallel_safe)]
+            pub fn #funcname_out #generics(input: #name #generics) -> &'static ::core::ffi::CStr {
                 let mut buffer = ::pgrx::stringinfo::StringInfo::new();
                 ::pgrx::inoutfuncs::JsonInOutFuncs::output(&input, &mut buffer);
-                buffer.into()
+                // SAFETY: We just constructed this StringInfo ourselves
+                unsafe { buffer.leak_cstr() }
```

Comment:
This is bad, for various reasons, but right now it has to be this way. The proper fix is blocked on https://github.com/pgcentralfoundation/pgrx/issues/1383 but this issue existing blocks solving that, because the previous, worse version of this code requires the staticizing hack and ignoring lifetimes in order to be allowed to ship. In other words:

**I know it's unsound, and what's currently extant is also unsound, so there's no actual change, except that now the unsoundness becomes fixable.**